### PR TITLE
INTYGFV-16671: Extended implementation of handle question to include …

### PR DESCRIPTION
**Pull request by mikeyOne**
…administrative messages

### CODE REVIEW

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/HandleMessageDomainService.java [hunk 1]
```diff
@@ -17,6 +18,45 @@ public class HandleMessageDomainService {
...
+      message.handle();
+    } else {
+      message.unhandle();
+    }
+
+    return messageRepository.save(message);
+  }
+
+  private static void deleteDraftAnswerIfExists(Message message) {
+    if (message.answer() != null && message.answer().status() == MessageStatus.DRAFT) {
```
Comment (ChrilleSigerhed) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/HandleMessageDomainService.java:53]: Just a thought, but for certificates, we are able to just call isDraft() function that returns true if draft, and false if not.

Should implement the same logic in answer?

```java
  public boolean isDraft() {
    return MessageStatus.DRAFT.equals(this.status);
  }
```

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/SetMessagesToHandleDomainService.java [hunk 2]
```diff
@@ -12,6 +13,10 @@ public class SetMessagesToHandleDomainService {
 
   public void handle(List<Message> messages) {
     messages.forEach(Message::handle);
+    messages.stream()
+        .filter(message -> message.answer() != null
+            && message.answer().status() == MessageStatus.DRAFT)
```
Comment (ChrilleSigerhed) [inline certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/SetMessagesToHandleDomainService.java:18]: Use isDraft function if we seem fit to implement it!

**Review by ChrilleSigerhed — APPROVED** [review]
Very good job, I think this looks great!

Left a minor suggestion regarding moving the status check into the Answer class, just so that we follow the same pattern as for Certificate!

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/HandleMessageDomainService.java (hunk 1 above)
Comment (mikeyOne) [reply certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/HandleMessageDomainService.java]: done

File: certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/SetMessagesToHandleDomainService.java (hunk 2 above)
Comment (mikeyOne) [reply certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/message/service/SetMessagesToHandleDomainService.java]: done

**Pull request merged**